pkg/metricscollector/v1beta1/file-metricscollector/file-metricscollector_test.go

```diff
@@ -93,28 +96,86 @@ func TestCollectObservationLog(t *testing.T) {
 				},
 			},
 		},
-		{
-			description: "Invalid file name",
-			filePath:    "invalid",
-			err:         true,
+		"Positive case for logs in TEXT format": {
+			filePath:   filepath.Join(testTextDataPath, "good.log"),
+			metrics:    []string{"accuracy", "loss"},
+			filters:    []string{"{metricName: ([\\w|-]+), metricValue: ((-?\\d+)(\\.\\d+)?)}"},
```

Comment:
As we mentioned in the issue, our motivation is to verify if this filter works fine to avoid bugs like this: https://github.com/kubeflow/katib/issues/1754

> We should add unit test for our [File Metrics Collector](https://github.com/kubeflow/katib/tree/master/pkg/metricscollector/v1beta1/file-metricscollector) to verify log parsing.
That will help to avoid issues similar to: https://github.com/kubeflow/katib/issues/1754

So, could you add more than valid and invalid cases?

Reply:
Yes, I'll do it later.

Reply:
Some new test cases similar to bugs in #1754 have been added to the test file.